.github/workflows/ci.yml

```
@@ -10,28 +10,22 @@
-
-    - name: Clone IDF Release From Github
-      uses: actions/checkout@v2
-      with:
-        repository: espressif/esp-idf
-        path: dependencies/idf-tools
-        submodules: 'true'
-        ref: release/v5.1
+    - uses: actions/checkout@v4
 
     - name: Set up Python
       uses: actions/setup-python@v2
```

actionlint (1.7.4)

the runner of "actions/setup-python@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)